app/controllers/medications_controller.rb

```diff
@@ -4,12 +4,23 @@ class MedicationsController < ApplicationController
   include CollectionPageSetupConcern
   include ReminderHelper
   include MedicationRefillHelper
+  include MedicationsHelper
   before_action :set_medication, only: %i[show edit update destroy]
+  respond_to :json, only: [:index]
```

Comment:
I don't understand why this is necessary since you still run index inside a respond_to block